libraries/src/Document/HtmlDocument.php

```diff
@@ -794,6 +798,39 @@ protected function _renderTemplate()
 			$with[] = $this->getBuffer($args['type'], $args['name'], $args['attribs']);
 		}
 
+		$this->_template = str_replace('</body>', self::renderIcons() . '</body>', $this->_template);
+
 		return str_replace($replace, $with, $this->_template);
 	}
+
+
+	/**
+	 * Renders the document icons
+	 *
+	 * @return  string  The svgs markup for the icons
+	 *
+	 * @since   4.0.0
+	 */
+	public function renderIcons()
+	{
+		$files        = [];
+
+		// Generate the file and load the stylesheet link
+		foreach ($this->icons as $key => $icon)
+		{
+			$file = HTMLHelper::image('vendor/font-awesome/' . $icon . '.svg', '', null,true, 1);
```

Comment:
We made something similar in DPCalendar for J3. While doing that, an idea popped up, to first search in the template for an svg and then in vendor. Like that a template can override the core icons.

Reply:
exactly what is happening here as well thanks to the magic of HTMLHelper::image